src/o365/spfx/commands/project/project-rename.ts

```diff
@@ -0,0 +1,193 @@
+      path.join(this.projectRootPath, 'config/deploy-azure-storage.json'),
+      path.join(this.projectRootPath, 'README.md')
+    ];
+
+    const replaceFileContent = (filePath: string) => {
+      if (fs.existsSync(filePath)) {
+        let existingContent = fs.readFileSync(filePath, 'utf-8');
+        let updatedContent;
+        if (filePath.endsWith('.json')) {
+          updatedContent = JSON.parse(existingContent);
+
+          if (updatedContent &&
+            updatedContent.name) {
+            updatedContent.name = args.options.newName;
+          }
+          if (updatedContent &&
+            updatedContent['@microsoft/generator-sharepoint'] &&
+            updatedContent['@microsoft/generator-sharepoint'].libraryName) {
+            updatedContent['@microsoft/generator-sharepoint'].libraryName = args.options.newName;
+          }
+          if (updatedContent &&
+            updatedContent['@microsoft/generator-sharepoint'] &&
+            updatedContent['@microsoft/generator-sharepoint'].solutionName) {
+            updatedContent['@microsoft/generator-sharepoint'].solutionName = args.options.newName;
+          }
+          if (updatedContent &&
+            updatedContent['@microsoft/generator-sharepoint'] &&
+            updatedContent['@microsoft/generator-sharepoint'].libraryId &&
+            args.options.generateNewId) {
+            updatedContent['@microsoft/generator-sharepoint'].libraryId = newId;
+          }
+          if (updatedContent &&
+            updatedContent.solution &&
+            updatedContent.solution.name) {
+            updatedContent.solution.name = updatedContent.solution.name.replace(new RegExp(projectName, 'g'), args.options.newName);
+          }
+          if (updatedContent &&
+            updatedContent.solution &&
+            updatedContent.solution.id &&
+            args.options.generateNewId) {
+            updatedContent.solution.id = newId;
+          }
+          if (updatedContent &&
+            updatedContent.paths &&
+            updatedContent.paths.zippedPackage) {
+            updatedContent.paths.zippedPackage = updatedContent.paths.zippedPackage.replace(new RegExp(projectName, 'g'), args.options.newName);
+          }
+          if (updatedContent &&
+            updatedContent.container) {
+            updatedContent.container = args.options.newName;
+          }
+          fs.writeFileSync(filePath, JSON.stringify(updatedContent, null, 2), 'utf-8');
+        } else {
+          updatedContent = existingContent.replace(new RegExp(projectName, 'g'), args.options.newName);
+          fs.writeFileSync(filePath, updatedContent, 'utf-8');
+        }
+        if (this.debug) {
+          cmd.log(`Updated ${filePath.split('/').pop()}`);
+        }
+      }
+    }
+
+    if (this.debug) {
+      cmd.log(`Renaming SharePoint Framework project to '${args.options.newName}'`);
+    }
+
+    filePaths.forEach((filePath) => {
+      replaceFileContent(filePath);
```

Comment:
Since each file that needs to be updates has a different structure, why call the same method on all of them? It would be more efficient and reliable to apply only the modifications relevant to the particular file and avoid any possible side-effects.

Reply:
Hi @waldekmastykarz I updated the **replaceFileContent** method structure to something like this. Let me know if this works:
```
const replaceFileContent = (filePath: string) => {
  if (!fs.existsSync(filePath)) {
    return;
  }
  const existingContent = fs.readFileSync(filePath, 'utf-8');
  let updatedContent;
  if (filePath.endsWith('.json')) {
    updatedContent = JSON.parse(existingContent);

    if (filePath.endsWith('package.json')) {
      //Update specific json properties
    }
    if (filePath.endsWith('.yo-rc.json')) {
      //Update specific json properties
    }
    if (filePath.endsWith('config/package-solution.json')) {
      //Update specific json properties
    }
    if (filePath.endsWith('config/deploy-azure-storage.json')) {
      //Update specific json properties
    }
    fs.writeFileSync(filePath, JSON.stringify(updatedContent, null, 2), 'utf-8');
  }
  if (filePath.endsWith('README.md')) {
    //Update content
  }
  if (this.debug) {
    cmd.log(`Updated ${filePath.split('/').pop()}`);
  }
}
```

Reply:
Hi @waldekmastykarz I've made all the requested changes along with the above mentioned one. Let me know if this works or any other changes are required. Thanks for your guidance! 😊

Reply:
Thank you! We'll take a look as quickly as possible 👍

Reply:
Hey @aakashbhardwaj619, what is the benefit of having one `replaceFileContent` method that uses separate conditional blocks for the different files it's called on, instead of separate methods for each file? It seems like the only common logic is reading file contents, parsing to JSON and writing the update structure to the file, so that could be separated out, but I think that keeping the different modifications separate would make it easier to maintain the code over time.

Reply:
Hi @waldekmastykarz you are right, no other benefit apart from the common logic you mentioned. I have now created separate methods for updating each of the files. So sorry that you had to review this PR multiple times.